src/main/java/com/myhome/controllers/PaymentController.java

```diff
@@ -62,11 +64,12 @@
       SchedulePaymentRequest request) {
     log.trace("Received schedule payment request");
 
-    return paymentService.getHouseMember(request.getMemberId())
-      .map(member -> communityService.findCommunityAdminById(request.getAdminId())
-                      .map(admin -> isUserAdminOfCommunityHouse(member.getCommunityHouse(), admin))
-                      .orElse(null))
-      .map(admin -> schedulePaymentApiMapper.schedulePaymentRequestToPaymentDto(request))
+    Optional<HouseMember> houseMember = paymentService.getHouseMember(request.getMemberId());
+    Optional<User> adminOptional = communityService.findCommunityAdminById(request.getAdminId());
+
+     return houseMember.map(member -> adminOptional.map(admin -> isUserAdminOfCommunityHouse(member.getCommunityHouse(), admin))
+                                                .orElse(null))
+      .map(success -> schedulePaymentApiMapper.enrichedSchedulePaymentRequestToPaymentDto(schedulePaymentApiMapper.enrichSchedulePaymentRequest(request, adminOptional.get(), houseMember.get())))
```

Comment:
`success` can be here false or true.
Can you explain this?

Reply:
I just chose that as a variable name to use the lambda expression, it means nothing

Reply:
I mean, what is the reason of calling `isUserAdminOfCommunityHouse` if the result is not processed

Reply:
Basically it returns a Boolean object true, or else null instead of false so that the map after that call doesn't run if the method returns null. I can do it a different way if you wish but if I return false the following map runs when it shouldn't

Reply:
Let's change the method `isUserAdminOfCommunityHouse` to:
```
  private Boolean isUserAdminOfCommunityHouse(CommunityHouse communityHouse, User admin) {
    return communityHouse.getCommunity().getAdmins().contains(admin);
  }
```

and use stream `.filter(/*new method*/)` here

Reply:
But if it doesn't contain the admin, it will return false which returns a Boolean object of false. Then the map function that called that function will return an Optional<Boolean> of that object, which means it has a value and so the next map function will be called. This is not what we want to happen. That's why if they're not an admin, return null which is in effect and empty optional object so the next map does not get called

Reply:
You can write it in this way:
```
return houseMember.flatMap(member -> adminOptional.filter(admin -> isUserAdminOfCommunityHouse(member.getCommunityHouse(), admin)))
      .map(success -> schedulePaymentApiMapper.enrichedSchedulePaymentRequestToPaymentDto(schedulePaymentApiMapper.enrichSchedulePaymentRequest(request, adminOptional.get(), houseMember.get())))
      .map(paymentService::schedulePayment)
      .map(schedulePaymentApiMapper::paymentToSchedulePaymentResponse)
      .map(response -> ResponseEntity.status(HttpStatus.CREATED).body(response))
      .orElseGet(() -> ResponseEntity.notFound().build());
```

Reply:
Thank you, that worked